### ISSUE TITLE
Options for ensembling forecasters 

### ISSUE THREAD

**Issue by ngupta23** [alan-turing-institute/sktime#583]
**Is your feature request related to a problem? Please describe.**

Currently ensemble forecasting model only supports mean reduction as [outlined here](https://github.com/alan-turing-institute/sktime/blob/30b06140eb6e94b2845e6239efd679414dc13188/sktime/forecasting/compose/_ensemble.py#L78). Can we add more options similar to [this](https://hcrystalball.readthedocs.io/en/latest/_modules/hcrystalball/ensemble/_simple_ensemble.html#SimpleEnsemble).

**Describe the solution you'd like**

Provide options to combine the forecasts using 

- [x] Median
- [x] Min/Max
- [ ] Weighted (currently defaults to `mean` which is equal weighting, but user could provide custom weighting for each model)

**Comment by ngupta23** [comment]
@mloning, I can add this if you are OK with it.

**Comment by mloning** [comment]
@ngupta23 yes, sounds good to me! What do you have in mind in terms of design? Adding a keyword argument to the constructor would perhaps be the simplest. Eventually, we could abstract the aggregation algorithm further.

**Comment by ngupta23** [comment]
I am thinking it would be similar to [this](https://hcrystalball.readthedocs.io/en/latest/_modules/hcrystalball/ensemble/_simple_ensemble.html#SimpleEnsemble) with an option for weighted

```
ensemble_func: {'mean', 'median', 'weighted', 'min', 'max'}
        Function to aggregate `base_learners` predictions
weights: Optional[Dict{<name>: <weight>}]
   None (default) will assume equal weight (i.e. same as mean)
```

**Comment by mloning** [comment]
@ngupta23 sorry for the late reply. Sounds good to me, would welcome a PR!

**Comment by aiwalter** [comment]
@ngupta23 I like your idea! I am asking myself if we then would also be able to tune those parameters of `EnsembleForecaster` also via `ForecastingGridSearchCV`? Maybe even tuning together with the other hyperparameters of the models contained in the `EnsembleForecaster`? This would however increase the number of fits quite a lot.

@mloning what do you think?

**Comment by fkiraly** [comment]
Remembering that we have already some architecture for weighted forecasting ensembles (see PR #333), I was wondering whether it would make sense to involve @magittan in the discussion to see how we integrate this with the current ensembling sub-module?

I think it would make a lot of sense to come up with a joint/consolidated design for forecasting ensembles that's flexible and composable - given that there already is one for weighted ensembles and on-line ensembles.

**Comment by fkiraly** [comment]
(might make sense to set up a phone call where @magittan gives a quick walkthrough through architecture of the weighted ensemble module?)

**Comment by fkiraly** [comment]
PS, @ngupta23: Min/Max is usually not very robust (extremely outlier sensitive), high/low quantiles are usually better

**Comment by mloning** [comment]
@aiwalter as long as we add the aggregation function to the constructor of the ensemble, tuning  should work 🙂

**Comment by magittan** [comment]
Sure, would be happy to participate in a phone call for what we have now for online-ensembling of forecasters.

**Comment by ngupta23** [comment]
> 
> 
> PS, @ngupta23: Min/Max is usually not very robust (extremely outlier sensitive), high/low quantiles are usually better

Yeah, I personally was more interested in the median (instead of mean) for the same reason. The only reason I added min/max there was due to the compatibility with [hcrystalball](https://hcrystalball.readthedocs.io/en/latest/_modules/hcrystalball/ensemble/_simple_ensemble.html#SimpleEnsemble). I would be OK if you decide not to include min/max.

**Comment by mloning** [comment]
Closed by #774
